src/lib.rs

```diff
@@ -240,23 +249,23 @@ mod tests {
     #[test]
     fn test_get_schema_name() {
         let fragment = r#"<Schema name="testname"></Schema>"#;
-        assert_eq!(Fragment::get_schema_name(fragment), Some("testname"));
+        assert_eq!(Fragment::get_schema_name(fragment).unwrap(), Some("testname"));
         let fragment = r#"<Cube name="testname"></Cube>"#;
-        assert_eq!(Fragment::get_schema_name(fragment), None);
+        assert_eq!(Fragment::get_schema_name(fragment).unwrap(), None);
     }
 
     #[test]
     fn test_get_share_dims() {
```

Comment:
Can you have separate tests for `SharedDimension` and then `Dimension` as a shared dimension? I think that right now it's just a test for `SharedDimension`?

Reply:
The Tests are of mixture some are for `Dimension` and few others are for `SharedDimension`. We can replicate the test for both instead of the other way.